src/components/ProgressButton.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FC, useState } from 'react';
+import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
+import ProgressBar from 'react-native-progress/Bar';
+
+import { LinearGradient } from 'app/components';
+import { typography, palette } from 'app/styles';
+
+interface Props {
+  timeoutMilis: number;
+  stepIntervalMilis: number;
+  onComplete: () => any;
+  title: string;
+  inProgressTitle: string;
+}
+
+const noIntervalID = -1;
+
+export const ProgressButton: FC<Props> = ({
+  timeoutMilis,
+  stepIntervalMilis,
+  onComplete,
+  title,
+  inProgressTitle,
+}: Props) => {
+  const [inProgress, setInProgress] = useState(false);
+  const [progress, setProgress] = useState(0);
+  const [intervalID, setIntervalID] = useState(noIntervalID);
+
+  const addWallet = () => {
```

Comment:
this button shouldn't be aware of a wallet, it should be a generic dump component

Reply:
done